pkgs/top-level/aliases.nix

```diff
@@ -1006,6 +1026,7 @@ mapAliases ({
   openimageio2 = openimageio; # Added 2023-01-05
   openimageio_1 = throw "'openimageio_1' has been removed, please update to 'openimageio' 2"; # Added 2023-06-14
   openisns = open-isns; # Added 2020-01-28
+  openjfx15 = javaPackages.openjfx15; # Added 2024-05-09
```

Comment:
Could you explain this please? I thought it is EOL too according to the commit message.

Reply:
JDK 16 needs it, and 16 is being kept as it isn't presently a maintenance burden.

Reply:
> JDK 16 needs it, and 16 is being kept as it isn't presently a maintenance burden.
> 
> Will admit that it is kinda weird to make it an alias right now though.

JDK 16 needs JFX 15? I thought all of these JFX vs JDK versions are matching.

Reply:
One would think :)


Thankfully it's just JDK 16 that is out of place, to my knowledge.

Reply:
> One would think :)
> 
> Thankfully it's just JDK 16 that is out of place, to my knowledge.

This is very non-trivial, perhaps this requires a comment.

Reply:
> Will admit that it is kinda weird to make it an alias right now though.

What do you mean by "right now" do you expect that in the future we will remove this attribute? When openjdk16 will be EOL as well? I'd vote for putting it in `all-packages.nix`, along with all others.

Reply:
Oh I remember why. Aliases aren't allowed to be used within Nixpkgs. I put it in aliases since the goal was to deprecate jfx 15 and openjdk 16 within Nixpkgs, but continue to export it for the sake of backwards compatibility. What I don't understand is why openjdk16 wasn't in all-packages.nix previously, while openjfx15 was.